File: gestion/seed/seed.js
// ════════════════════════════════════════════════
// Script de seed — à exécuter UNE SEULE FOIS depuis un ordinateur,
// jamais depuis le site lui-même (utilise une clé d'administration).
//
// Ce script :
//  1. Crée un compte Firebase Authentication pour chacun des 14 membres
//     (identifiant "prenom@cptsdbmb.local" + mot de passe par défaut)
//  2. Initialise les documents Firestore des missions/actions et un
//     agenda de démarrage (mêmes données que le mode démo)
//
// Utilisation :
//   1. Déposer le fichier de clé de service (Firebase > Paramètres du
//      projet > Comptes de service > Générer une nouvelle clé privée)
//      sous le nom "serviceAccountKey.json" dans le dossier gestion/
//   2. npm install firebase-admin   (dans ce dossier gestion/seed)
//   3. node seed.js
// ════════════════════════════════════════════════
const admin = require('firebase-admin');
const path = require('path');
const serviceAccount = require(path.join(__dirname, '..', 'serviceAccountKey.json'));

admin.initializeApp({ credential: admin.credential.cert(serviceAccount) });
const auth = admin.auth();
const db = admin.firestore();

const EMAIL_DOMAIN = '@cptsdbmb.local';

// Même liste que gestion/data.js — mot de passe par défaut à transmettre à chacun.
const MEMBERS = [
  { id: 'patrick',        name: 'Dr Patrick Laugareil',       password: 'Patrick2026!' },
  { id: 'samia',          name: 'Dr Samia Ben Ayad-Beloufa',  password: 'Samia2026!' },
  { id: 'olivier',        name: 'Dr Olivier Barclay',          password: 'Olivier2026!' },
  { id: 'malika.kemache', name: 'Malika Kemache',              password: 'MalikaK2026!' },
  { id: 'frederic',       name: 'Frédéric Mourad',             password: 'Frederic2026!' },
  { id: 'fatima',         name: 'Dr Fatima Bargui',            password: 'Fatima2026!' },
  { id: 'lounes',         name: 'Lounès Kemmache',             password: 'Lounes2026!' },
  { id: 'sabrina',        name: 'Sabrina Ben Ayad',            password: 'Sabrina2026!' },
  { id: 'solene',         name: 'Solène Dias',                 password: 'Solene2026!' },
  { id: 'celia',          name: 'Célia Vilus',                 password: 'Celia2026!' },
  { id: 'sighane',        name: 'Sighane Diop',                password: 'Sighane2026!' },
  { id: 'francine',       name: 'Francine Braflan',            password: 'Francine2026!' },
  { id: 'malika.mouchon', name: 'Malika Mouchon',              password: 'MalikaM2026!' },
  { id: 'mathilde',       name: 'Mathilde Moysan',             password: 'Mathilde2026!' },
];

// Chargé depuis gestion/data.js (module ES) pour rester en phase avec l'application.
async function loadMissions() {
  const mod = await import('../data.js');
  return mod.MISSIONS;
}

async function seedMembers() {
  for (const m of MEMBERS) {
    const email = m.id + EMAIL_DOMAIN;
    try {
      const existing = await auth.getUserByEmail(email).catch(() => null);
      if (existing) {
        console.log(`- Déjà existant : ${m.name} (${email})`);
        continue;
      }
      await auth.createUser({ email, password: m.password, displayName: m.name });
      console.log(`+ Compte créé : ${m.name} (${email} / ${m.password})`);
    } catch (e) {
      console.error(`! Erreur pour ${m.name} :`, e.message);
    }
  }
}

async function seedActions() {
  const missions = await loadMissions();
  const batch = db.batch();
  let count = 0;
  missions.forEach(m => m.actions.forEach(a => {
    const ref = db.collection('actions').doc(`${m.id}__${a.id}`);
    batch.set(ref, {
      indicateurs: a.indicateurs.map(() => false),
      livrables: a.livrables.map(() => false),
      remarque: '',
      updatedBy: '',
      updatedAt: null,
    }, { merge: true });
    count++;
  }));
  await batch.commit();
  console.log(`+ ${count} actions initialisées dans Firestore.`);
}

async function seedEvents() {
  const existing = await db.collection('events').limit(1).get();
  if (!existing.empty) { console.log('- Des événements existent déjà, aucun ajouté.'); return; }
  const today = new Date();
  const in5 = new Date(today); in5.setDate(in5.getDate() + 5);
  const in12 = new Date(today); in12.setDate(in12.getDate() + 12);
  await db.collection('events').add({
    title: "Soirée de formation — Diabète", date: in5.toISOString().slice(0,10), time: '19:00',
    description: "À destination de tous les professionnels de santé du territoire.",
    createdBy: 'Mathilde Moysan', createdAt: admin.firestore.FieldValue.serverTimestamp(),
  });
  await db.collection('events').add({
    title: "Réunion de concertation — Ménopause", date: in12.toISOString().slice(0,10), time: '18:30',
    description: "Réunion pluriprofessionnelle de partage d'expériences.",
    createdBy: 'Dr Olivier Barclay', createdAt: admin.firestore.FieldValue.serverTimestamp(),
  });
  console.log('+ 2 événements de démarrage ajoutés.');
}

(async () => {
  console.log('=== Création des comptes ===');
  await seedMembers();
  console.log('\n=== Initialisation des missions/actions ===');
  await seedActions();
  console.log('\n=== Initialisation de l\'agenda ===');
  await seedEvents();
  console.log('\nTerminé. Transmettez à chaque membre son identifiant (prénom) et son mot de passe ci-dessus.');
  process.exit(0);
})();

File: gestion/seed/seed.py
# ════════════════════════════════════════════════
# Script de seed (Python, équivalent de seed.js) — à exécuter UNE SEULE FOIS.
# Crée les 14 comptes Firebase Authentication + initialise Firestore.
# Utilise gestion/serviceAccountKey.json (jamais committé, voir .gitignore).
# ════════════════════════════════════════════════
import os
import firebase_admin
from firebase_admin import credentials, auth, firestore

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
cred = credentials.Certificate(os.path.join(BASE_DIR, "serviceAccountKey.json"))
firebase_admin.initialize_app(cred)
db = firestore.client()

EMAIL_DOMAIN = "@cptsdbmb.local"

MEMBERS = [
    ("patrick",        "Dr Patrick Laugareil",       "Patrick2026!"),
    ("samia",          "Dr Samia Ben Ayad-Beloufa",  "Samia2026!"),
    ("olivier",        "Dr Olivier Barclay",          "Olivier2026!"),
    ("malika.kemache", "Malika Kemache",              "MalikaK2026!"),
    ("frederic",       "Frédéric Mourad",             "Frederic2026!"),
    ("fatima",         "Dr Fatima Bargui",            "Fatima2026!"),
    ("lounes",         "Lounès Kemmache",             "Lounes2026!"),
    ("sabrina",        "Sabrina Ben Ayad",            "Sabrina2026!"),
    ("solene",         "Solène Dias",                 "Solene2026!"),
    ("celia",          "Célia Vilus",                 "Celia2026!"),
    ("sighane",        "Sighane Diop",                "Sighane2026!"),
    ("francine",       "Francine Braflan",            "Francine2026!"),
    ("malika.mouchon", "Malika Mouchon",              "MalikaM2026!"),
    ("mathilde",       "Mathilde Moysan",             "Mathilde2026!"),
]

# (missionId, actionId, nb_indicateurs, nb_livrables) — doit rester en phase avec gestion/data.js
ACTIONS = [
    ("acces-soins", "medecin-traitant", 6, 7),
    ("acces-soins", "snp-sas", 3, 5),
    ("acces-soins", "telemedecine", 2, 6),
    ("parcours-patient", "parcours-cardio", 4, 5),
    ("parcours-patient", "parcours-sante-mentale", 2, 6),
    ("parcours-patient", "parcours-sorties-hospit", 2, 4),
    ("prevention", "depistage-cancers", 4, 5),
    ("prevention", "journees-prevention", 3, 4),
    ("crise-sanitaire", "plan-crise", 2, 3),
    ("qualite-soins", "reunions-concertation", 1, 3),
    ("qualite-soins", "soirees-formation", 2, 4),
    ("accompagnement-ps", "securite-soignants", 2, 6),
    ("accompagnement-ps", "accueil-stagiaires", 2, 4),
    ("accompagnement-ps", "aide-installation", 2, 4),
]


def seed_members():
    print("=== Création des comptes ===")
    for member_id, name, password in MEMBERS:
        email = member_id + EMAIL_DOMAIN
        try:
            existing = auth.get_user_by_email(email)
            print(f"- Déjà existant : {name} ({email})")
        except auth.UserNotFoundError:
            auth.create_user(email=email, password=password, display_name=name)
            print(f"+ Compte créé : {name} ({email} / {password})")


def seed_actions():
    print("\n=== Initialisation des missions/actions ===")
    batch = db.batch()
    for mission_id, action_id, n_ind, n_liv in ACTIONS:
        ref = db.collection("actions").document(f"{mission_id}__{action_id}")
        batch.set(ref, {
            "indicateurs": [False] * n_ind,
            "livrables": [False] * n_liv,
            "remarque": "",
            "updatedBy": "",
            "updatedAt": None,
        }, merge=True)
    batch.commit()
    print(f"+ {len(ACTIONS)} actions initialisées dans Firestore.")


def seed_events():
    print("\n=== Initialisation de l'agenda ===")
    existing = list(db.collection("events").limit(1).stream())
    if existing:
        print("- Des événements existent déjà, aucun ajouté.")
        return
    import datetime
    today = datetime.date.today()
    in5 = today + datetime.timedelta(days=5)
    in12 = today + datetime.timedelta(days=12)
    db.collection("events").add({
        "title": "Soirée de formation — Diabète", "date": in5.isoformat(), "time": "19:00",
        "description": "À destination de tous les professionnels de santé du territoire.",
        "createdBy": "Mathilde Moysan", "createdAt": firestore.SERVER_TIMESTAMP,
    })
    db.collection("events").add({
        "title": "Réunion de concertation — Ménopause", "date": in12.isoformat(), "time": "18:30",
        "description": "Réunion pluriprofessionnelle de partage d'expériences.",
        "createdBy": "Dr Olivier Barclay", "createdAt": firestore.SERVER_TIMESTAMP,
    })
    print("+ 2 événements de démarrage ajoutés.")


if __name__ == "__main__":
    seed_members()
    seed_actions()
    seed_events()
    print("\nTerminé. Transmettez à chaque membre son identifiant (prénom) et son mot de passe ci-dessus.")
